qemu/tests/multi_disk_wild_hotplug.py

```diff
@@ -0,0 +1,112 @@
+"""Wild hot-plug-unplug test"""
+import copy
+import logging
+import time
+import os
+
+from virttest.qemu_monitor import QMPCmdError
+from virttest import error_context, env_process
+from virttest import data_dir as virttest_data_dir
+
+
+# This decorator makes the test function aware of context strings
+@error_context.context_aware
+def run(test, params, env):
+    """
+    1) Boot the vm with multiple data disks.
+    2) Run some operation in guest if request.
+    3) Execute device_del for data disks.
+    4) Sleep some time.
+    5) Execute device_add for data disks.
+    6) Sleep some time.
+    7) repeat step 3-6 if request.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def _run_sg_luns():
+        file_name = "guest_sg_luns.sh"
+        guest_dir = "/tmp/"
+        deps_dir = virttest_data_dir.get_deps_dir()
+        host_file = os.path.join(deps_dir, file_name)
+        guest_file = guest_dir + file_name
+        vm.copy_files_to(host_file, guest_dir)
+        session.sendline("$SHELL " + guest_file)
+
+    def _configure_images_params():
+        for i in range(image_num):
+            name = "stg%d" % i
+            params['image_name_%s' % name] = "images/%s" % name
+            params['image_size_%s' % name] = stg_image_size
+            params["images"] = params["images"] + " " + name
+            if params["drive_format"] == "scsi-hd":
+                params["drive_bus_%s" % name] = 1
+                params["blk_extra_params_%s" % name] = "lun=%d" % i
+            image_params = params.object_params(name)
+            env_process.preprocess_image(test, image_params, name)
+
+    def _get_images_params():
+        for i in range(image_num):
+            name = "stg%d" % i
+            dev = vm.devices.get_by_qid(name)[0]
+            images_params[name] = copy.deepcopy(dev.params)
+
+    def _hotplug_images():
+        for i in range(1, image_num):
+            name = "stg%d" % i
+            try:
+                vm.monitor.cmd("device_add", images_params[name], debug=False)
+            except QMPCmdError as e:
+                logging.warning('Ignore hotplug error: %s', str(e))
```

Comment:
After adding arm support, I hit the following error. It is suspected that the insertion is too fast.
Survey
`23:10:15 WARNI| Ignore hotplug error: QMP command 'device_add' failed    (arguments: OrderedDict([('driver', 'virtio-blk-pci'), ('id', 'stg1'), ('drive', 'drive_stg1'), ('bootindex', '2'), ('write-cache', 'on'), ('bus', 'pcie-root-port-5'), ('addr', '0x0'), ('iothread', 'iothread0')]),    error message: {'class': 'GenericError', 'desc': "Duplicate ID 'stg1' for device"})`

Reply:
This error can be ignored for virtio_blk due to the timeout hard to determind. and virtio_blk is not the test point. 
I suggest you mentioned scenario just added under virtio_scsi